src/query.ts

```diff
@@ -1131,6 +1131,7 @@ export class Query<
         errors: [
           {
             errorCode: r.sf__Error,
+            statusCode: r.sf__Error,
```

Comment:
these are bulk the bulk v2 API, I don't see `statusCode` there:
https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_failed_results.htm?q=__error

Reply:
Hey Cristian, thanks for the review!

I made this change in anticipation of the `errorCode` being a deprecated field, but from your other comment, it seems some APIs return this.

I first noticed this when creating a permSetAssignment record using the following command in jsForce:

```typescript
return await org.sobject('PermissionSetAssignment').create(permissionAssignments);
```

^^ this returned the error in my initial PS post above.

To further investigate this, I went ahead and made the same request via the Rest API and it actually returned error code instead of status code 😆 :
```
MALFORMED_ID: PermissionSet ID: id value of incorrect type: 0
    at HttpApi.getError (/Users/aaron.csetter/repos/sandbox/node_modules/@jsforce/jsforce-node/lib/http-api.js:290:56)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/aaron.csetter/repos/sandbox/node_modules/@jsforce/jsforce-node/lib/http-api.js:127:33
    at async main2 (/Users/aaron.csetter/repos/sandbox/dist/index.js:36:17) {
  data: {
    message: 'PermissionSet ID: id value of incorrect type: 0',
    errorCode: 'MALFORMED_ID',
    fields: [ 'PermissionSetId' ]
  },
  errorCode: 'MALFORMED_ID'
}
```

I haven't looked into what underlying API that `Connection.sobject()` is using to create records, but it's likely NOT the rest API based on what you and I are seeing, haha.

Reply:
Hey @cristiand391 

Looks like when multiple records are passed in the the `Connection.create()` method, the request uses [sObject Collections](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_composite_sobjects_collections_create.htm), which even the documentation says `statusCode` is returned in the error: